src/main/java/lotto/domain/LottoGame.java

```diff
@@ -1,58 +1,55 @@
 package lotto.domain;
 
 import lotto.strategy.AutoLottoStrategy;
+import lotto.strategy.LottoStrategy;
 
+import java.util.EnumMap;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
 public class LottoGame {
     private LottoTickets tickets;
-    private Map<Rank, Integer> results;
 
     public LottoGame(LottoTickets tickets) {
         this.tickets = tickets;
-        initResults();
     }
 
-    public LottoGame(LottoTickets tickets, Map<Rank, Integer> results) {
-        this.tickets = tickets;
-        this.results = results;
-    }
-
-
     public LottoTickets getTickets() {
         return tickets;
     }
-    
-    public Map<Rank, Integer> getResults() {
-        return results;
-    }
 
-    public void generateAutoLottoNumbers(AutoLottoStrategy autoLottoStrategy) {
-        tickets.generateAllLottoNumbers(autoLottoStrategy);
-    }
 
-    private void initResults() {
-        results = new HashMap<Rank, Integer>();
+    private Map<Rank, Integer> initializeResults() {
+        // Rank의 모든 값을 초기화하여 결과 맵 생성
+        Map<Rank, Integer> results = new EnumMap<>(Rank.class);
         for (Rank rank : Rank.values()) {
             results.put(rank, 0);
         }
+        return results;
     }
 
-    public void gameStart(List<Integer> winningNumbers) {
-        for (LottoTicket ticket : tickets.getLottoTickets()) {
-            int matchCount = ticket.matchLottoNumbers(winningNumbers);
-            Rank rank = Rank.getRankByMatchCount(matchCount);
-            results.put(rank, results.get(rank) + 1);
-        }
+    public GameResult gameStart(List<Integer> winningNumbers) {
+        Map<Rank, Integer> results = initializeResults();
+
+        // 각 티켓의 당첨 등수를 계산하여 결과 맵에 반영
+        tickets.getLottoTickets().stream()
```

Comment:
비즈니스 로직에서 getter로 값을 꺼내기 보다 메시지를 보내 등수별 당첨 결과를 받으면 어떨까요?
```suggestion
        Map<Rank, Integer> results = tickets.match(winningNumbers);
```

당첨 번호의 타입도 List 대신 LottoTicket 타입을 활용해 보셔도 좋을 것 같아요.